protocol/v2/ssv/msgqueue/signed_msg.go

```diff
@@ -30,7 +28,7 @@ func signedMsgIndexValidator(msg *spectypes.SSVMessage) *specqbft.SignedMessage
 	if msg == nil {
 		return nil
 	}
-	if msg.MsgType != spectypes.SSVConsensusMsgType && msg.MsgType != message.SSVDecidedMsgType {
+	if msg.MsgType != spectypes.SSVConsensusMsgType {
```

Comment:
add k.D check

Reply:
there's no `k.D` here, the `msg.MsgType != spectypes.SSVConsensusMsgType` check covers decided messages as well